thrust/thrust/system/cuda/detail/future.inl

```diff
@@ -916,9 +924,10 @@ public:
 
 namespace detail
 {
+_CCCL_SUPPRESS_DEPRECATED_PUSH // for thrust::optional
 
-template <typename X, typename Deleter>
-_CCCL_HOST optional<unique_stream> try_acquire_stream(int, std::unique_ptr<X, Deleter>&) noexcept
+  template <typename X, typename Deleter>
+  _CCCL_HOST optional<unique_stream> try_acquire_stream(int, std::unique_ptr<X, Deleter>&) noexcept
```

Comment:
I would rather just use `cuda::std::optional` here

Reply:
I pushed a change for that

Reply:
That's what I tried, but you need C++17 for that :D. Also, this PR will be backported to 2.8 where we only have C++11 as well.

Reply:
Discussed offline: I removed the commits to change to `cuda::std::optional`. I will add them as a follow-up PR.